### DISCUSSION TITLE
direct linker output?

### DISCUSSION THREAD

**Discussion by tengai650** [microsoft/vscode-cmake-tools#3830] [General]
is there a way to control where the linker places the DLL or .EXE? Any examples?

Thanks,

Patrick

**Comment by tengai650** [comment]
Ok I believe I found a solution and applies to 'multi-config' generators context (e.g. Visual Studio, Ninja Multi-Config).

I'm doing the following:

```
foreach(OUTPUTCONFIG ${CMAKE_CONFIGURATION_TYPES})
    string(TOUPPER ${OUTPUTCONFIG} OUTPUTCONFIG)
    set(CMAKE_RUNTIME_OUTPUT_DIRECTORY_${OUTPUTCONFIG} ${CMAKE_SOURCE_DIR}/bin/${OUTPUTCONFIG}/plugins)
endforeach(OUTPUTCONFIG CMAKE_CONFIGURATION_TYPES)
```

[Footnote](https://stackoverflow.com/questions/6594796/how-do-i-make-cmake-output-into-a-bin-dir)
[Footnote](https://cmake.org/cmake/help/latest/variable/CMAKE_RUNTIME_OUTPUT_DIRECTORY_CONFIG.html#cmake-runtime-output-directory-config)

**Comment by bobbrow** [comment]
For my projects, I use cmake [INSTALL](https://cmake.org/cmake/help/latest/command/install.html) to have cmake copy a successful build to a location relative to `CMAKE_INSTALL_PREFIX`. Then I don't worry about where the intermediate files end up.

CMakeLists.txt
```
install(
  TARGETS myTarget
  DESTINATION bin
)

```
